python/core/auto_generated/qgsmaplayer.sip.in

```diff
@@ -503,6 +503,18 @@ Returns new instance of QgsMapLayerRenderer that will be used for rendering of g
     virtual QgsRectangle extent() const;
 %Docstring
 Returns the extent of the layer.
+%End
+
+    QgsRectangle geographicExtent( bool actual = false ) const;
```

Comment:
```suggestion
    QgsRectangle geographicExtent( bool forceRecalculate = false ) const;
```

Reply:
Actually can we rename this from geographicExtent to wgs84Extent? WGS84 is a poor choice of CRS for many reasons, and I'd like to see us avoid reinforcing the misconception that WGS84 === "the correct choice for geographic coordinates".

Reply:
Just to explain where the naming comes from: there is a geographicExtent tag in WMS getCapabilities and all this work was triggered by the finding that the conversion was taking a long time on the server and we could cut the times by pre-calculating the extent and storing it in the project.

Reply:
Right, I get that -- I just don't think we have to directly mirror that naming in the QgsMapLayer API and can instead use something a bit more explicit. The fact that we equate geographic extent to WGS84 extent is a server specific detail really.

Reply:
>  and I'd like to see us avoid reinforcing the misconception that WGS84 === "the correct choice for geographic coordinates"

Indeed you're right. I'm taking a look.